boards: arm: b_l4s5i_iot01a and stm32h7b3i_dk: fix OSPI flash size

GeorgeCGV opened the pull request:
The OSPI flash driver expects size in bits. This PR uses bits instead of bytes for the size.

Fixes #52691.

GeorgeCGV committed:
boards: arm: b_l4s5i_iot01a: fix OSPI flash size

The OSPI flash driver expects size in bits.
Use bits instead of bytes for the size.

Signed-off-by: Georgij Cernysiov <geo.cgv@gmail.com>

erwango commented on boards/arm/b_l4s5i_iot01a/b_l4s5i_iot01a.dts:
```diff
@@ -248,7 +248,7 @@ zephyr_udc0: &usbotg_fs {
 		reg = <0>;
 		ospi-max-frequency = <DT_FREQ_M(26)>; /* for Voltage Range 2 */
-		size = <DT_SIZE_M(8)>; /* 64 Megabits = 8 Megabytes */
+		size = <DT_SIZE_M(64)>; /* 64 Mebibit */
```
s/Mebibit/Megabits

erwango replied:
s/Mebibit/Megabits

FRASTM commented:
Seems that stm32h7b3i_dk.dts too is  declaring size of the mx25lm51245 nor octoflash in MBytes :
`size = <DT_SIZE_M(64)>;`  --> `size = <DT_SIZE_M(512)>;`

GeorgeCGV commented:
> Seems that stm32h7b3i_dk.dts too is declaring size of the mx25lm51245 nor octoflash in MBytes : `size = <DT_SIZE_M(64)>;` --> `size = <DT_SIZE_M(512)>;`
> 
> That could be fixed here in another commit ?

can be done here

GeorgeCGV committed:
boards: arm: stm32h7b3i_dk: fix OSPI flash size

The OSPI flash driver expects size in bits.
Use bits instead of bytes for the size.

Signed-off-by: Francois Ramu <francois.ramu@st.com>
Signed-off-by: Georgij Cernysiov <geo.cgv@gmail.com>

FRASTM reviewed (approved):
Thanks
